packages/react/src/components/Slider/Slider.tsx

```diff
@@ -0,0 +1,276 @@
+import { handleRef, Ref } from '@stardust-ui/react-component-ref'
+import cx from 'classnames'
+
+import {
+  applyAccessibilityKeyHandlers,
+  AutoControlledComponent,
+  ChildrenComponentProps,
+  commonPropTypes,
+  isFromKeyboard,
+  partitionHTMLProps,
+  UIComponentProps,
+  RenderResultConfig,
+  setWhatInputSource,
+} from '../../lib'
+import {
+  ComponentEventHandler,
+  ShorthandValue,
+  WithAsProp,
+  withSafeTypeForAs,
+  Omit,
+} from '../../types'
+import { Accessibility } from '../../lib/accessibility/types'
+import { sliderBehavior } from '../../lib/accessibility'
+import { SupportedIntrinsicInputProps } from '../../lib/htmlPropsUtils'
+import Box from '../Box/Box'
+
+const defaultPropValues = {
+  max: 100,
+  min: 0,
+  step: 1,
+  value: 50,
+}
+
+const processInputValues = (
+  p: Pick<SliderProps, 'min' | 'max'> & Pick<SliderState, 'value'>,
+): { min: number; max: number; value: number; valueAsPercentage: string } => {
+  let min = _.toNumber(p.min)
+  let max = _.toNumber(p.max)
+  let value = _.toNumber(p.value)
+
+  if (isNaN(min)) min = defaultPropValues.min
+  if (isNaN(max)) max = defaultPropValues.max
+  value = isNaN(value) ? defaultPropValues.value : Math.min(max, Math.max(min, value))
+  const valueAsPercentage = `${(100 * (value - min)) / (max - min)}%`
+
+  return { min, max, value, valueAsPercentage }
+}
+
+export interface SliderSlotClassNames {
+  input: string
+  inputWrapper: string
+  rail: string
+  thumb: string
+  track: string
+}
+
+export interface SliderProps
+  extends UIComponentProps,
+    ChildrenComponentProps,
+    Omit<SupportedIntrinsicInputProps, 'defaultValue'> {
+  /**
+   * Accessibility behavior if overridden by the user.
+   * @default sliderBehavior
+   */
+  accessibility?: Accessibility
+
+  /** The default value of the slider. */
+  defaultValue?: React.ReactText
+
+  /**
+   * A slider can be read-only and unable to change states.
+   * @default false
+   */
+  disabled?: SupportedIntrinsicInputProps['disabled']
+
+  /**
+   * A slider can take the width of its container.
+   * @default false
+   */
+  fluid?: boolean
+
+  /**
+   * Callback that creates custom accessibility message a screen reader narrates when the value changes.
+   * @param {SliderProps} props - Slider props.
+   */
+  getA11yValueMessageOnChange?: (props: SliderProps) => string
+
+  /** Shorthand for the input component. */
+  input?: ShorthandValue
+
+  /** Ref for input DOM node. */
+  inputRef?: React.Ref<HTMLElement>
+
+  /**
+   * The maximum value of the slider.
+   * @default 100
+   */
+  max?: SupportedIntrinsicInputProps['max']
+
+  /**
+   * The minimum value of the slider
+   * @default 0
+   */
+  min?: SupportedIntrinsicInputProps['min']
+
+  /**
+   * Called after item checked state is changed.
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All props.
+   */
+  onChange?: ComponentEventHandler<SliderProps>
```

Comment:
```suggestion
  onChange?: ComponentEventHandler<SliderProps & { value: string }>
```

I believe that this is more accurate typing

Reply:
Basically for all components, the right type is `Props & State`. Would suggest addressing this for all components in a separate PR

Reply:
I don't agree with the `onChange?: ComponentEventHandler<SliderProps & { value: string }>` suggestion. `value` can be number as well. Also `value` is part of `SliderProps` so I don't see what the issue is exactly. Can I resolve?
@mnajdova @layershifter

Reply:
For this component this is definitely not a blocker, but it is something that we will need to tackle for other components.

Reply:
Value in callback can't be `number`, please check it again.
https://github.com/stardust-ui/react/pull/1559#discussion_r302952170

Reply:
indeed, @layershifter is right; applied change but as @mnajdova says we need to align these interfaces in a separate PR; thanks guys, resolving 😃